docs/DeveloperGuide.md

```diff
@@ -97,20 +97,39 @@ Given below is the Sequence Diagram for interactions within the `Logic` componen
 
 ![Structure of the Model Component](images/ModelClassDiagram.png)
 
-**API** : [`Model.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/model/Model.java)
+**API** : [`Model.java`](https://github.com/AY2021S2-CS2103-T14-1/tp/blob/master/src/main/java/seedu/address/model/Model.java)
 
 The `Model`,
 
 * stores a `UserPref` object that represents the user’s preferences.
-* stores the address book data.
-* exposes an unmodifiable `ObservableList<Person>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
-* does not depend on any of the other three components.
+* stores a `CommandHistory` object that represents all previous commands entered by the user.
+* stores an `AliasMapping` object that represents the mapping of aliases to actual commands.
+* stores the SunRez data in a `StatefulAddressBook`.
+    * which stores a list of `ReadOnlyAddressBook` objects, each representing a state after a state-changing command is executed.  
+* exposes the following unmodifiable `ObservableList<T>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list changes.
+    * `ObservableList<Resident>`
+    * `ObservableList<Room>`
+    * `ObservableList<Issue>`
+* does not depend on any of the other three components (`Storage`, `Logic`, `UI`).
 
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** An alternative (arguably, a more OOP) model is given below. It has a `Tag` list in the `AddressBook`, which `Person` references. This allows `AddressBook` to only require one `Tag` object per unique `Tag`, instead of each `Person` needing their own `Tag` object.<br>
-![BetterModelClassDiagram](images/BetterModelClassDiagram.png)
+The above diagram does not go into detail on how the specific sub-packages interact with each other. 
+This is due to both PlantUML diagramming restrictions as well as an intention to provide a high-level visualisation at the start.
 
-</div>
+The section below zooms in a **little bit** on how the `ModelManager` and `StatefulAddressBook` interact with the lower level sub-packages.
+Finer details than what is shown in the section below can be seen under the implementation section of each of the models.
+
+#### Zoomed in view of Room, Resident and ResidentRoom 
+![Zoomed in view of Room, Resident and ResidentRoom](images/high-level-models/ResidentRoomZoomIn.png)
```

Comment:
The zoomed in view for the 3 here is rather clustered, is there anyway to reduce the clustering?